gk/bpf.c

```diff
@@ -156,6 +201,12 @@ gk_bpf_encapsulate(struct gk_bpf_pkt_ctx *ctx, int priority)
 	if (unlikely(priority < 0 || priority > PRIORITY_MAX))
 		return -EINVAL;
 
+	if (priority == PRIORITY_GRANTED && ctx->direct_if_possible) {
```

Comment:
1. `direct_if_possible` should be a new parameter of `gk_bpf_encapsulate()` instead of a field in `struct gk_bpf_pkt_ctx` because we want BPF programs to decide when they want direct delivery.

2. It reads more natural to test that `direct_if_possible != 0` before `priority == PRIORITY_GRANTED`.